core/lib/spree/testing_support/dummy_app.rb

```diff
@@ -47,35 +47,33 @@ def self.setup(gem_root:, lib_name:, auto_migrate: true)
 
   class Application < ::Rails::Application
     config.load_defaults("#{Rails::VERSION::MAJOR}.#{Rails::VERSION::MINOR}")
-    config.eager_load = false
-    config.cache_classes = true
-    config.cache_store = :memory_store
-    config.serve_static_assets = true
-    config.public_file_server.headers = { 'Cache-Control' => 'public, max-age=3600' }
-    config.whiny_nils = true
-    config.consider_all_requests_local = true
-    config.action_controller.allow_forgery_protection = false
-    config.action_controller.default_protect_from_forgery = false
-    config.action_controller.perform_caching = false
-    config.action_dispatch.show_exceptions = false
-    config.active_support.deprecation = :stderr
-    config.action_mailer.delivery_method = :test
-    config.active_support.deprecation = :stderr
-    config.secret_key_base = 'SECRET_TOKEN'
-
-    config.action_mailer.delivery_job = "ActionMailer::MailDeliveryJob" if RAILS_6_OR_ABOVE
-    config.action_mailer.preview_path = File.expand_path('dummy_app/mailer_previews', __dir__)
-    config.action_mailer.show_previews = true
+    config.cache_classes = true # mimic production
+    config.eager_load = false # suite performance
+    config.public_file_server.enabled = true # as we don't use a web server, let Rails serve them
+    config.public_file_server.headers = { 'Cache-Control' => 'public, max-age=3600' } # suite performance
+    config.cache_store = :memory_store # suite performance
+    config.consider_all_requests_local = true # makes debugging easier
+    config.action_controller.allow_forgery_protection = false # mimic production
+    config.action_controller.default_protect_from_forgery = false # mimic production
+    config.action_controller.perform_caching = true # mimim production
+    config.action_dispatch.show_exceptions = false # makes debugging easier
+    config.active_support.deprecation = :stderr # makes debugging easier
+    config.log_level = :debug # makes debugging easier
+    config.action_mailer.delivery_method = :test # don't try to send mails
+    config.secret_key_base = 'SECRET_TOKEN' # no need to use credentials file
+    config.action_mailer.preview_path = File.expand_path('dummy_app/mailer_previews', __dir__) # preview path within the dummy app
+    config.action_mailer.perform_caching = false # mimic production
+    config.i18n.fallbacks = true # mimic production
```

Comment:
There's a lot of duplication in the reasoning for these. I also don't think appending all these comments to these lines is very readable. What if we grouped them by the reasoning? Something like:

```suggestion
    # Make the test environment more production-like:
    config.cache_classes = true
    config.action_controller.allow_forgery_protection = false
    config.action_controller.default_protect_from_forgery = false
    config.action_controller.perform_caching = true
    config.action_mailer.perform_caching = false
    config.i18n.fallbacks = true

    # Make debugging easier:
    config.consider_all_requests_local = true
    config.action_dispatch.show_exceptions = false
    config.active_support.deprecation = :stderr
    config.log_level = :debug

    # Improve test suite performance:
    config.eager_load = false
    config.public_file_server.headers = { 'Cache-Control' => 'public, max-age=3600' }
    config.cache_store = :memory_store

    # We don't use a web server, so we let Rails serve assets.
    config.public_file_server.enabled = true

    # We don't want to send email in the test environment.
    config.action_mailer.delivery_method = :test

    # No need to use credentials file in a test environment.
    config.secret_key_base = 'SECRET_TOKEN'

    # Set the preview path within the dummy app:
    config.action_mailer.preview_path = File.expand_path('dummy_app/mailer_previews', __dir__)
```

Reply:
I preferred to group by scope, but we can do either way